### PR TITLE
[GEN-1844]: refactor "status" component

### PR DESCRIPTION
**Pull request by BenElferink**
This pull request includes several updates to the `frontend/webapp` components and reusable components to improve type safety, refactor status components, and enhance the styling of various elements. The most important changes include the introduction of new status components, refactoring of existing components to use these new status components, and updates to the `PlatformTypes` enum.

### Type Safety and Enum Updates:
* [`frontend/webapp/types/actions.ts`](diffhunk://#diff-3ae6790a1d85d899bad68bedc63fd62bd18aeda31add61046a76187d9175d21dR3-R7): Added `PlatformTypes` enum to replace string literals for platform types.

### Component Refactoring:
* [`frontend/webapp/components/main/header/cp-title/index.tsx`](diffhunk://#diff-ec1cadaa336f57dd2899b60bfc2c46359e847d6a289f3828d1f47e806a74c28fR4-R8): Updated `PlatformTitle` component to use `PlatformTypes` enum and refactored code for better readability. [[1]](diffhunk://#diff-ec1cadaa336f57dd2899b60bfc2c46359e847d6a289f3828d1f47e806a74c28fR4-R8) [[2]](diffhunk://#diff-ec1cadaa336f57dd2899b60bfc2c46359e847d6a289f3828d1f47e806a74c28fL31-R39)
* [`frontend/webapp/components/main/header/index.tsx`](diffhunk://#diff-2c96f91ec30d2116981a9c0a562820ff9fd87c8292cb5dca11a45d6fb2ac6c04R3-R26): Updated `MainHeader` component to use `PlatformTypes` enum and replaced `Status` with `ConnectionStatus`. [[1]](diffhunk://#diff-2c96f91ec30d2116981a9c0a562820ff9fd87c8292cb5dca11a45d6fb2ac6c04R3-R26) [[2]](diffhunk://#diff-2c96f91ec30d2116981a9c0a562820ff9fd87c8292cb5dca11a45d6fb2ac6c04L42-R40)

### Status Component Enhancements:
* [`frontend/webapp/reuseable-components/status/index.tsx`](diffhunk://#diff-8828fa39eed82424bd382e5bd240ee5efc607723527e6dc649d92a3d0c32aed3L3-L100): Refactored the `Status` component to include additional props and improved styling. Exported new `ActiveStatus` and `ConnectionStatus` components.
* [`frontend/webapp/reuseable-components/status/active-status/index.tsx`](diffhunk://#diff-9485e5934f1643bf9cae8c13ff4ea216eb2e1767bb7ccdb7ed6b6ac3a41d8efeR1-R8): Created new `ActiveStatus` component extending `Status` with specific props.
* [`frontend/webapp/reuseable-components/status/connection-status/index.tsx`](diffhunk://#diff-11d70e0e626f4e9e67f4ca7da05f0bba5a6b28cd20f0fd793daefcc719a7e61fR1-R8): Created new `ConnectionStatus` component extending `Status` with specific props.

### Divider Component Update:
* [`frontend/webapp/reuseable-components/divider/index.tsx`](diffhunk://#diff-2505266b420a0b2e50b6f2026d9d00fd42ef705dd54103de84f7e2e0e2221812R3-L29): Updated `Divider` component to accept a `type` prop for conditional styling based on notification type.

These changes collectively improve the maintainability and readability of the codebase, enhance type safety, and provide a more consistent styling approach across the application.

### CODE REVIEW

File: frontend/webapp/components/main/header/index.tsx [hunk 1]
```diff
@@ -40,7 +40,7 @@ export const MainHeader: React.FC<MainHeaderProps> = () => {
       <AlignLeft>
         <Image src='/brand/transparent-logo-white.svg' alt='logo' width={84} height={20} />
         <PlatformTitle type='k8s' />
-        {!connecting && <Status title={title} subtitle={message} isActive={active} withIcon withBackground />}
+        {!connecting && <ConnectionStatus title={title} subtitle={message} isActive={active} />}
```
Comment (alonkeyval) [inline frontend/webapp/components/main/header/index.tsx:43]: we can use k8s as const from consts file

File: frontend/webapp/components/main/header/index.tsx [hunk 2]
```diff
@@ -3,7 +3,7 @@ import Image from 'next/image';
 import styled from 'styled-components';
 import { PlatformTitle } from './cp-title';
 import { useConnectionStore } from '@/store';
-import { Status } from '@/reuseable-components';
+import { ConnectionStatus } from '@/reuseable-components';
 import { NotificationManager } from '@/components/notification';
 
 interface MainHeaderProps {}
```
Comment (alonkeyval) [inline frontend/webapp/components/main/header/index.tsx:9]: we can now use our new common style to make this component smaller:


export const HeaderContainer = styled(FlexRow)`
  width: 100%;
  padding: 12px 0;
  background-color: ${({ theme }) => theme.colors.darker_grey};
  border-bottom: 1px solid rgba(249, 249, 249, 0.16);
`;

export const HeaderSection = styled(FlexRow)`
  margin: 0 32px;
  flex: 1;
  &:first-child {
    margin-right: auto;
  }
  &:last-child {
    margin-left: auto;
  }
`;

File: frontend/webapp/reuseable-components/status/active-status/index.tsx [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import React from 'react';
+import { Status, type StatusProps } from '@/reuseable-components';
+
+interface Props extends StatusProps {}
+
+export const ActiveStatus: React.FC<Props> = ({ isActive, ...props }) => {
+  return <Status title={isActive ? 'Active' : 'Inactive'} isActive={isActive} {...props} />;
```
Comment (alonkeyval) [inline frontend/webapp/reuseable-components/status/active-status/index.tsx:7]: do we need two more files for active and connection status if they only hold the status component?

File: frontend/webapp/reuseable-components/status/index.tsx [hunk 4]
```diff
@@ -1,100 +1,103 @@
...
-            <TextDivider $isActive={isActive} />
-            <SubTitle $isActive={isActive} $withSpecialFont={withSpecialFont} $withSmaller={withSmaller}>
-              {subtitle}
-            </SubTitle>
-          </TextWrapper>
-        )}
-      </TextWrapper>
+          {!!subtitle && (
+            <TextWrapper>
+              <Divider orientation='vertical' length={`${size - 2}px`} color={isPale ? undefined : (isActive ? theme.text.success : theme.text.error) + hexPercentValues['030']} />
```
Comment (alonkeyval) [inline frontend/webapp/reuseable-components/status/index.tsx:93]: handle the style inside the styled component block to reduce the theme path's dependency.

File: frontend/webapp/reuseable-components/status/active-status/index.tsx (hunk 3 above)
Comment (BenElferink) [reply frontend/webapp/reuseable-components/status/active-status/index.tsx:7]: `Status` is a base component, it has a variety of style options, ranging from colors, fonts, borders, and backgrounds.

`ActiveStatus`, `ConnectionStatus`, and `InstrumentStatus` all leverage `Status`, but apply different styles.

In the case of `ActiveStatus`, we have 1 version with icon + border, and 1 version with nothing.
In the case of `ConnectionStatus`, we have 1 version with icon + background.
In the case of `InstrumentStatus `, we have 1 version with icon + border, and "pale" if inactive.

So yes, we need these to be separated.

**Review by alonkeyval — APPROVED** [review]

**Pull request merged**